### ISSUE TITLE
Logout user if refresh fails?

### ISSUE THREAD

**Issue by dobromir-hristov** [websanova/vue-auth#152]
Hey. Today I was messing around and I got into a strange case where I got into a "white screen of death" scenario. If I have a corrupted token, or my refresh request returns a 400 error - bad request, you cant do anything until you manually clean up your localstorage token. 
Should we do something like this, and is this a good way of doing it?
```js
function _refreshPerform(data) {
      data.error = (error) => {
        if (!error.OK) {
          this.options.logoutPerform.call(this, data);
        }
      }
      __duckPunch.call(this, 'refresh', data);
    }
```
I tried adding this error callback to my `refreshData` setup in `main.,js` but I dont have access to the `logout` method, nor the `$auth` property at all for that matter. 
Also I used a fat arrow function to get `this` to stay because it was getting lost. Please correct me if I'm wrong.

**Comment by websanova** [comment]
What version are you using?

**Comment by dobromir-hristov** [comment]
@websanova/vue-auth@2.10.2-beta

**Comment by websanova** [comment]
Ok, so actually this handled in the driver.

https://github.com/websanova/vue-auth/blob/master/drivers/http/vue-resource.1.x.js

You can override this for now. But yes probably this should be improved to
catch all errors.

On Apr 30, 2017 23:43, "Dobromir Hristov" <notifications@github.com> wrote:

@websanova/vue-auth@2.10.2-beta

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/websanova/vue-auth/issues/152#issuecomment-298239525>,
or mute the thread

**Comment by dobromir-hristov** [comment]
Tried the `_invalidToken` method and that does not fire at all. Did you mean anywhere else ?

**Comment by websanova** [comment]
ok, yes, you are right. I've just put a fix in for this in `2.11.0-beta`. Let me know if that works out.

**Comment by dobromir-hristov** [comment]
Like a charm. 
```js
  _invalidToken: function (res) {
    if (res.status === 401 || res.status === 400) {
      this.options.logoutProcess.call(this, res, { redirect: this.options.authRedirect })
    }
  },
```
